service/src/main/java/greencity/service/ubs/UBSManagementServiceImpl.java

```diff
@@ -1680,12 +1680,17 @@ private void checkUpdateResponsibleEmployeeDto(UpdateAllOrderPageDto updateAllOr
     @Override
     public AddBonusesToUserDto addBonusesToUser(AddBonusesToUserDto addBonusesToUserDto,
         Long orderId) {
-        User currentUser = userRepository.findUserByOrderId(orderId)
-            .orElseThrow(() -> new UserNotFoundException(USER_WITH_CURRENT_ID_DOES_NOT_EXIST));
         Order order = orderRepository.findById(orderId)
             .orElseThrow(() -> new NotFoundException(ORDER_WITH_CURRENT_ID_DOES_NOT_EXIST + orderId));
+        CounterOrderDetailsDto prices = getPriceDetails(orderId);
+        Long overpayment = calculateOverpayment(order, setTotalPrice(prices).longValue());
+        if (overpayment == 0L) {
+            throw new BadRequestException(USER_HAS_NO_OVERPAYMENT);
+        }
+        User currentUser = userRepository.findUserByOrderId(orderId)
```

Comment:
It doesn't need to find the user in DB, you can get him from Order.
```suggestion
        User currentUser = order.getUser();
```